src/main/java/ch/njol/skript/expressions/ExprChunk.java

```diff
@@ -18,83 +18,117 @@
+		);
 	}
-	
-	@SuppressWarnings("null")
+
+	private int pattern;
 	private Expression<Location> locations;
-	
-	@SuppressWarnings({"unchecked", "null"})
+	private Expression<World> worlds;
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		if (matchedPattern == 0) {
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		pattern = matchedPattern;
+		if (pattern == 0) {
 			locations = (Expression<Location>) exprs[1];
-			if (exprs[0] != null)
+			if (exprs[0] != null) {
 				locations = Direction.combine((Expression<? extends Direction>) exprs[0], locations);
-		} else {
+			}
+		} else if (pattern == 1) {
 			locations = (Expression<Location>) exprs[0];
+		} else {
+			worlds = ((Expression<World>) exprs[0]);
 		}
-		setExpr(locations);
 		return true;
 	}
-	
-	@Override
-	protected Chunk[] get(final Event e, final Location[] source) {
-		return get(source, Location::getChunk);
-	}
-	
-	@Override
-	public Class<? extends Chunk> getReturnType() {
-		return Chunk.class;
-	}
-	
+
 	@Override
-	public String toString(final @Nullable Event e, final boolean debug) {
-		return "the chunk at " + locations.toString(e, debug);
+	protected @Nullable Chunk[] get(Event event) {
+		if (pattern != 2) {
+			return locations.stream(event)
+				.map(Location::getChunk)
+				.toArray(Chunk[]::new);
+		}
+		World[] worlds = this.worlds.getArray(event);
+		if (worlds.length == 0)
+			return new Chunk[0];
+		List<Chunk> chunks = new ArrayList<>();
+		for (World world : worlds) {
+			chunks.addAll(Arrays.asList(world.getLoadedChunks()));
+		}
+		return chunks.toArray(new Chunk[0]);
 	}
-	
+
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
+	public Class<?>[] acceptChange(ChangeMode mode) {
 		if (mode == ChangeMode.RESET)
 			return new Class[0];
 		return null;
 	}
-	
+
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
+	@SuppressWarnings("deprecation")
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
 		assert mode == ChangeMode.RESET;
-		
-		final Chunk[] cs = getArray(e);
-		for (final Chunk c : cs)
-			c.getWorld().regenerateChunk(c.getX(), c.getZ());
+		Chunk[] chunks = get(event);
+		for (Chunk chunk : chunks)
+			chunk.getWorld().regenerateChunk(chunk.getX(), chunk.getZ());
+	}
+
+	@Override
+	public boolean isSingle() {
+		if (pattern == 2)
+			return false;
+		return locations.isSingle();
+	}
+
+	@Override
+	public Class<? extends Chunk> getReturnType() {
+		return Chunk.class;
 	}
 	
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		if (pattern == 2)
+			return "the loaded chunks of " + worlds.toString(event, debug);
+		return "the chunk at " + locations.toString(event, debug);
```

Comment:
```suggestion
		return "chunk at " + locations.toString(event, debug);
```
Skript can add `the` in some context already for printing.